console/network/src/lib.rs

```diff
@@ -132,6 +133,12 @@ pub trait Network:
     const MAX_FEE: u64 = 1_000_000_000_000_000;
     /// The maximum number of microcredits that can be spent on a finalize block.
     const TRANSACTION_SPEND_LIMIT: u64 = 100_000_000;
+    /// The maximum number of microcredits that can be spent by the transactions in a batch.
+    /// This implies the block spend limit is bounded at `TRANSACTION_SPEND_LIMIT * N::NUM_MAX_CERTIFICATES`.
+    const BATCH_SPEND_LIMIT: u64 = Self::TRANSACTION_SPEND_LIMIT;
+    /// The base cost in microcredits to verify an execution.
+    /// NOTE: this constant reflects the compute cost of an execution, but is not required to be paid by the user.
+    const EXECUTION_BASE_COST: u64 = 2_000_000; // 2 million microcredits.
```

Comment:
"is not required to be paid by the user" What does this mean? 

A `transfer_public` is currently around 51k microcredits, why is there a need for the additional 2 million microcredits?

Reply:
Is this to arbitrarily say that a proposal is upper bounded by `BATCH_SPEND_LIMIT` / `EXECUTION_BASE_COST` number of executions?

Reply:
Executions do incur compute, we just don't price it in at the moment. So it makes sense to take it into account when determining the spend limit - or better wording: 'compute limit'.

Right now it doesn't matter too much because we have a batch proposal size limit, we could leave it it.

However, in the future we may want to consider getting rid of the batch proposal size limit, and only keep the compute limit, at which point we better not forget that proof verification takes up a significant amount of compute.

I agree 2 credits seems like a lot - but note that single-threaded proof verification costs multiple milliseconds. I think in any case this can likely be lowered to 1 or 0.5 for now.

Reply:
It seems like `cost` is an overloaded term here, as it implies it's used for the execution fees paid by the user.

In theory, the execution fee should already price in the cost of compute. An execution fee is the cost of finalize + storage cost; the storage cost can be considered the base level of payment for storing and performing the verification on the transaction. 

It feels confusing that a constant is tacked on here to limit the proposal spend by misrepresenting and inflating the "cost" of each transaction.

Reply:
I agree it is confusing. Just for the record and for anyone reading this: execution (snark proof) verification and its compute is currently not priced in.

However, I realized yesterday that actually execution verification is no longer on the hot path, due to this PR merged this week: https://github.com/ProvableHQ/snarkOS/pull/3451

So thank you for flagging, I think we can remove this "virtual proof verification cost". The spend limit's only aim is to limit the amount of compute in the block generation hot path.

Reply:
Removed in [ed5b491](https://github.com/ProvableHQ/snarkVM/pull/2601/commits/ed5b491c2018b2b4ee4e2f55ded088e72df4f088).